ddspipe_core/src/cpp/core/DdsPipe.cpp

```diff
@@ -221,47 +223,121 @@ void DdsPipe::discovered_endpoint_(
         const Endpoint& endpoint) noexcept
 {
     std::lock_guard<std::mutex> lock(mutex_);
+    discovered_endpoint_nts_(endpoint);
+}
+
+void DdsPipe::updated_endpoint_(
+        const Endpoint& endpoint) noexcept
+{
+    std::lock_guard<std::mutex> lock(mutex_);
+    updated_endpoint_nts_(endpoint);
+}
+
+void DdsPipe::removed_endpoint_(
+        const Endpoint& endpoint) noexcept
+{
+    std::lock_guard<std::mutex> lock(mutex_);
+    removed_endpoint_nts_(endpoint);
+}
 
+void DdsPipe::discovered_endpoint_nts_(
+        const Endpoint& endpoint) noexcept
+{
     logDebug(DDSPIPE, "Endpoint discovered in DDS Pipe core: " << endpoint << ".");
 
-    // Set as discovered only if the endpoint is a Reader
-    // If non Readers in topics, it is considered as non discovered
-    if (endpoint.is_reader())
+    if (RpcTopic::is_service_topic(endpoint.topic) && endpoint.is_reader() && endpoint.is_server_endpoint())
+    {
+        // Service server discovered
+        discovered_service_nts_(RpcTopic(
+                endpoint.topic), endpoint.discoverer_participant_id, endpoint.guid.guid_prefix());
+    }
+    else if (is_endpoint_relevant_nts_(endpoint))
```

Comment:
Watch out, this would create a bridge for a RPC client I believe (discovering reply reader).